### PR TITLE
Add gpaw

### LINKED ISSUE #7834: Add gpaw
I'm interested in seeing (an mpi version of) [gpaw](https://wiki.fysik.dtu.dk/gpaw) on conda-forge. I think the only piece missing is libxc.
I see that that the psi4 team has a recipe for it:
https://github.com/psi4/psi4meta/blob/master/conda-recipes/libxc/meta.yaml

@loriab do you mind if I make a PR to staged-recipes based on your recipe for libxc?

### PR DESCRIPTION
**Pull request by bjodah**
Let's see how this goes.

Closes #7834

### CODE REVIEW

**Comment by bjodah** [comment]
Ready for review.

File: recipes/gpaw/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - 001_config-py.patch
```
Comment (marcelotrevisani) [inline recipes/gpaw/meta.yaml]: Please, could you write an explanation why do you need this patch?

File: recipes/gpaw/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+build:
+  number: {{ build }}
+  string: {{ build_string }}
```
Comment (marcelotrevisani) [inline recipes/gpaw/meta.yaml]: Why do you need this build string?

Comment (bjodah) [reply recipes/gpaw/meta.yaml]: I'm basically duplicating the recipe for LAMMPS which is also built with and without MPI:
https://github.com/conda-forge/lammps-feedstock/blob/master/recipe/meta.yaml
I just thought it was needed for the conda packages to have different names depending on MPI imlementation...

File: recipes/gpaw/meta.yaml (hunk 1 above)
Comment (bjodah) [reply recipes/gpaw/meta.yaml]: Sure, should I add it as a comment in `meta.yaml`? (the explanation is that gpaw's config.py assumes libpython is a dynamic library, whereas on conda-forge it seems to be a static library).

**Comment by bjodah** [comment]
Maybe @jan-janssen (the author of the lammps recipe) wouldn't mind taking a look? (apologies Jan if you find my ping annoying).

**Comment by jan-janssen** [comment]
Looks good to me, I am just wondering if it makes sense to also support `mpich`.

**Comment by bjodah** [comment]
I was planning on trying to add mpich in a future pull-request to the feedstock. I started with openmpi because I know it's supposed to work, and I've had problems with other MPI implementations in the past (but things might have changed since). I can try to make it work here already if that's preferable?

**Comment by mariusvniekerk** [comment]
I'd do the mpich work in the feedstock

**Comment by mariusvniekerk** [comment]
cc @isuruf

File: recipes/gpaw/meta.yaml (hunk 1 above)
Comment (bjodah) [reply recipes/gpaw/meta.yaml]: I've opened a merge request upstream: https://gitlab.com/gpaw/gpaw/merge_requests/525

**Comment by bjodah** [comment]
Anything else you want me to address here before we merge?

**Comment by bjodah** [comment]
@isuruf would you mind giving this a quick look? (and thanks for your work on blas2!)

File: recipes/gpaw/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    - ase
+  run:
+    - {{ mpi }}  # [mpi != 'nompi']
+    - python
+    - libxc
+    - libblas
+    - libcblas
+    - scalapack
+    - fftw
+    - numpy
```
Comment (chrisburr) [inline recipes/gpaw/meta.yaml]: Is this needed here? [docs](https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=numpy#building-against-numpy)
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/gpaw/conda_build_config.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+python:
```
Comment (chrisburr) [inline recipes/gpaw/conda_build_config.yaml]: I might be missing something but could `python` be removed from here and `skip: True  # [py<36]` be added to the `build` section of `meta.yaml` instead?

File: recipes/gpaw/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    - gpaw test -j 1 --reverse --from generic/mixer.py                          # [mpi == 'nompi']
+    - test -x $PREFIX/bin/gpaw-python                                           # [mpi != 'nompi']
+    - export OMPI_MCA_plm=isolated                                              # [mpi == 'openmpi']
+    - export OMPI_MCA_btl_vader_single_copy_mechanism=none                      # [mpi == 'openmpi']
+    - export OMPI_MCA_rmaps_base_oversubscribe=yes                              # [mpi == 'openmpi']
+    - mpiexec -np 1 gpaw-python -m gpaw test --reverse --from generic/mixer.py  # [mpi != 'nompi']
+
+about:
+  home: https://wiki.fysik.dtu.dk/gpaw
+  license: GPL v3
```
Comment (chrisburr) [inline recipes/gpaw/meta.yaml]: From https://opensource.org/licenses/alphabetical:
```suggestion
  license: GPL-3.0
```

**Comment by bjodah** [comment]
Thank you for the review!

**Pull request merged**